packages/telemetry/src/base/client/azure-appinsight-client.ts

```diff
@@ -0,0 +1,186 @@
+     *
+     * @param eventName Categorize the type of the event within the scope of an extension.
+     * @param properties A set of string properties to be reported
+     * @param measurements  A set of numeric measurements to be reported
+     * @param sampleRate Sampling the event to be sent
+     * @param telemetryHelperProperties Properties that are passed to specific TelemetryClient for generating specific properties (E.g. ToolsSuiteTelemetryClient)
+     * @param ignoreSettings Ignore telemetryEnabled settings and skip submitting telemetry data
+     */
+    public async report(
+        eventName: string,
+        properties: { [key: string]: string | boolean },
+        measurements: { [key: string]: number },
+        sampleRate: SampleRate | undefined,
+        telemetryHelperProperties?: { [key: string]: string },
+        ignoreSettings?: boolean
+    ): Promise<void> {
+        if ((ignoreSettings !== undefined && !ignoreSettings) || !TelemetrySettings.telemetryEnabled) {
+            return;
+        }
+        const { client, event } = this.prepareClientAndEvent(eventName, properties, measurements, sampleRate);
+        this.trackEvent(client, event);
+    }
+
+    /**
+     * Provide specification of telemetry event to be sent.
+     *
+     * @param eventName Categorize the type of the event within the scope of an extension.
+     * @param properties A set of string properties to be reported
+     * @param measurements A set of numeric measurements to be reported
+     * @param sampleRate  Sampling the event to be sent
+     * @returns TelemetryClient instance and telemetry event
+     */
+    private prepareClientAndEvent(
+        eventName: string,
+        properties: { [key: string]: string | boolean },
+        measurements: { [key: string]: number },
+        sampleRate = SampleRate.NoSampling
+    ): { client: appInsights.TelemetryClient; event: appInsights.Contracts.EventTelemetry } {
+        const client = this.clients.get(sampleRate) as appInsights.TelemetryClient;
+
+        const eventHeader: EventHeader = new EventHeader(this.extensionName, eventName);
+        const event: appInsights.Contracts.EventTelemetry = {
+            name: eventHeader.toString(),
+            properties: properties as Record<string, string>,
+            measurements: measurements
+        };
+
+        return {
+            client,
+            event
+        };
+    }
+
+    /**
+     * Send telemetry event in blocking style. It blocks
+     * the subsequent statements until telemetry event has been sent.
+     *
+     * @param client TelemetryClient instance
+     * @param event Telemetry event
+     * @returns Promise<void>
+     */
+    private async trackEventBlocking(
+        client: appInsights.TelemetryClient,
+        event: appInsights.Contracts.EventTelemetry
+    ): Promise<void> {
+        return new Promise((resolve) => {
+            if (process.env.SAP_UX_FIORI_TOOLS_DISABLE_TELEMETRY !== 'true') {
```

Comment:
Is this expected behaviour, the promise will be in a pending state?

Reply:
resolve is inside callback setting of `client.flush` call. But I missed resolve for the else branch. Added fixed.